spring/spring-boot-3/remote/src/test/java/io/micrometer/core/instrument/binder/cache/NotAuthorizedRemoteCacheMetricBinderTest.java

```diff
@@ -0,0 +1,109 @@
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class NotAuthorizedRemoteCacheMetricBinderTest {
+
+   @RegisterExtension
+   static InfinispanServerExtension infinispanServerExtension = InfinispanServerExtensionBuilder.server();
+
+   private RemoteCache<String, String> cacheAdminConnection;
+   private RemoteCache<String, String> cacheObserverConnection;
+
+   private MeterRegistry registry = new SimpleMeterRegistry();
+   private CacheMeterBinder binder;
+
+   @BeforeEach
+   void bindToRegistry() {
+      this.binder = this.binder();
+      this.binder.bindTo(this.registry);
+   }
+
+   public CacheMeterBinder binder() {
+      org.infinispan.configuration.cache.ConfigurationBuilder cacheConfigBuilder =
+            new org.infinispan.configuration.cache.ConfigurationBuilder();
+      cacheConfigBuilder.clustering().cacheMode(CacheMode.DIST_SYNC);
+      cacheConfigBuilder.security().authorization().roles("admin");
+
+      StringConfiguration stringConfiguration =
+            new StringConfiguration(cacheConfigBuilder.build().toStringConfiguration("mycache"));
+
+      ConfigurationBuilder clientBuilder = new ConfigurationBuilder();
+      clientBuilder.statistics().enable();
+      clientBuilder.clientIntelligence(ClientIntelligence.BASIC);
+      clientBuilder.security()
+            .authentication()
+            .username(TestUser.ADMIN.getUser())
+            .password(TestUser.ADMIN.getPassword());
+
+      RemoteCacheManager remoteCacheManagerAdmin =
+            infinispanServerExtension.hotrod().withClientConfiguration(clientBuilder)
+                                     .createRemoteCacheManager();
+      cacheAdminConnection = remoteCacheManagerAdmin.administration().getOrCreateCache("mycache", stringConfiguration);
+
+
+      clientBuilder = new ConfigurationBuilder();
+      clientBuilder.statistics().enable();
+      clientBuilder.clientIntelligence(ClientIntelligence.BASIC);
+      clientBuilder.security()
+            .authentication()
+            .username(TestUser.OBSERVER.getUser())
+            .password(TestUser.OBSERVER.getPassword());
+
+      RemoteCacheManager remoteCacheManagerObserver =
+            infinispanServerExtension.hotrod().withClientConfiguration(clientBuilder)
+                  .createRemoteCacheManager();
+
+      // The cache is got with OBSERVER instead of ADMIN
+      cacheObserverConnection = remoteCacheManagerObserver.getCache("mycache");
+      RemoteInfinispanCacheMeterBinderProvider remoteInfinispanCacheMeterBinderProvider =
+            new RemoteInfinispanCacheMeterBinderProvider();
+      return (CacheMeterBinder) remoteInfinispanCacheMeterBinderProvider
+            .getMeterBinder(new SpringCache(cacheObserverConnection), emptyList());
+   }
+
+   @Test
+   void size() {
+      assertThatThrownBy(() -> this.registry.get("cache.size")).isInstanceOf(MeterNotFoundException.class);
```

Comment:
You have to change this, as it will no longer throw an exception

Reply:
yes, I pushed all now